optaplanner-wb-screens/optaplanner-wb-solver-editor/optaplanner-wb-solver-editor-client/src/main/java/org/optaplanner/workbench/screens/solver/client/editor/TerminationTreeItemContent.java

```diff
@@ -224,315 +240,335 @@ public HTMLElement getElement() {
+                                      }
+
+                                      @Override
+                                      public void setNewValue() {
+                                          view.setMillisecondsSpent(0l);
+                                          model.setMillisecondsSpentLimit(0l);
+                                          view.setSecondsSpent(0l);
+                                          model.setSecondsSpentLimit(0l);
+                                          view.setMinutesSpent(MINUTES_SPENT_DEFAULT_VALUE);
+                                          model.setMinutesSpentLimit(MINUTES_SPENT_DEFAULT_VALUE);
+                                          view.setHoursSpent(0l);
+                                          model.setHoursSpentLimit(0l);
+                                          view.setDaysSpent(0l);
+                                          model.setDaysSpentLimit(0l);
+                                      }
+
+                                      @Override
+                                      public void setLabelStrings() {
+                                          view.setFormLabelText(translationService.getTranslation(SolverEditorConstants.TerminationTreeItemContentTimeSpent));
+                                          view.setFormLabelHelpContent(translationService.getTranslation(SolverEditorConstants.TerminationTreeItemContentTimeSpentHelp));
+                                      }
+                                  });
+        terminationManagerMap.put(UNIMPROVED_MILLISECONDS_SPENT_LIMIT,
+                                  new TerminationManager() {
+                                      @Override
+                                      public void hideViewInputs() {
+                                          view.hideUnimprovedTimeSpentInput();
+                                      }
+
+                                      @Override
+                                      public void removeModelValue() {
+                                          model.setUnimprovedMillisecondsSpentLimit(null);
+                                          model.setUnimprovedSecondsSpentLimit(null);
+                                          model.setUnimprovedMinutesSpentLimit(null);
+                                          model.setUnimprovedHoursSpentLimit(null);
+                                          model.setUnimprovedDaysSpentLimit(null);
+                                      }
+
+                                      @Override
+                                      public void setExistingValue(Object value) {
+                                          value = terminationConfigForm.getTerminationValue(model,
+                                                                                            UNIMPROVED_MILLISECONDS_SPENT_LIMIT);
+                                          view.setUnimprovedMillisecondsSpent(value == null ? 0l : Long.valueOf(value.toString()));
```

Comment:
Just a minor readability comment: It is recommended to use `L` instead of `l` to signify `long` literals, because they are too easily confused with `1`.